src/coreclr/jit/optimizer.cpp

```diff
@@ -6374,21 +6381,22 @@ void Compiler::optRecordSsaUses(GenTree* tree, BasicBlock* block)
 // Arguments:
 //   origExpr - tree to hoist
 //   exprBb   - block containing the tree
-//   lnum     - loop that we're hoisting origExpr out of
+//   loop     - loop that we're hoisting origExpr out of
 //
-void Compiler::optPerformHoistExpr(GenTree* origExpr, BasicBlock* exprBb, unsigned lnum)
+void Compiler::optPerformHoistExpr(GenTree* origExpr, BasicBlock* exprBb, FlowGraphNaturalLoop* loop)
 {
     assert(exprBb != nullptr);
+    assert(loop->EntryEdges().size() == 1);
 
+    BasicBlock* preheader = loop->EntryEdges()[0]->getSourceBlock();
```

Comment:
A nit, but I wonder if it would be nice to have convenience functions to get a particular edge/etc. from a vector, e.g.
```suggestion
    BasicBlock* preheader = loop->EntryEdge(0)->getSourceBlock();
    
FlowEdge* EntryEdge(unsigned i)
{
    return m_entryEdges[i];
}
```